controllers/secret_cert_controller.go

```diff
@@ -104,28 +107,58 @@ func (r *SecretCertReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	log.Info("checking secret type is tls")
+	log.Info("checking secret type is kubernetes.io/tls or Opaque")
 
-	if desired.Type != secretType {
-		// it's not for us
-		return ctrl.Result{}, nil
-	}
+	var (
+		tlsKey []byte
+		tlsCrt []byte
+	)
 
-	log.Info("ensuring tls.key is present")
+	switch desired.Type {
+	case TLSSecretType:
+		log.Info("ensuring tls.key is present")
 
-	tlsKey, ok := desired.Data["tls.key"]
-	if !ok {
-		// cert doesn't exist yet
-		log.Info("missing tls.key, we don't care about it yet")
-		return ctrl.Result{}, nil
-	}
+		ok := false
+		tlsKey, ok = desired.Data["tls.key"]
+
+		if !ok {
+			// cert doesn't exist yet
+			log.Info("missing tls.key, we don't care about it yet")
+			return ctrl.Result{}, nil
+		}
+
+		log.Info("ensuring tls.crt is present")
+
+		tlsCrt, ok = desired.Data["tls.crt"]
+
+		if !ok {
+			// cert doesn't exist yet
+			log.Info("missing tls.crt, we don't care about it yet")
+			return ctrl.Result{}, nil
+		}
+	case opaqueSecretType:
+		var (
+			ok                        bool
+			publicKeyFieldName        = "public-key"
+			publicKeyEnabledFieldName = "public-key-enabled"
+		)
+		// public-key-enabled field indicates that this resource will be tracked by the secret controller.
+		enabled, exists := desired.Data[publicKeyEnabledFieldName]
```

Comment:
Since we do not have certain field names to store public key (for example, kubernete.io/tls types use `tls.crt` & `tls.key`), we are getting a field name where the cert-controller should check for public key. If users provide an invalid field name, the secret object will not have the public key in that field. Therefore, i've thought that we should check the existence of the public key.

Reply:
If "public-key" is not a standard field for a secret resource, couldn't we just check that the public key exists?

Reply:
Yes `public-key` is not standard field for a secret resource. I just want to prevent checking all Opaque secrets by secret-controller via checking `enabled` field. If users provide `enabled` field, secret-controller checks that secret. Otherwise, it is not managed by cert-controller as before. If you think it is not necessary, we can remove that part.